### PR TITLE
Avoid deadlock in the `ls_parse.py` script

### PR DESCRIPTION
**Pull request by harmim**
Python `subprocess.Popen.wait` deadlocks when using `stdout=subprocess.PIPE`, and the child process generates enough output to a pipe such that it blocks waiting for the OS pipe buffer to accept more data. See https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait.

Spotted while trying to compile Xen `4.11`.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by harmim** [comment]
@tautschnig @karkhaz Can you please review this?

**Review by karkhaz — APPROVED** [review]
LGTM, thank you for this fix. I'll leave it to a maintainer to trigger the checks.

**Review by tautschnig — APPROVED** [review]

**Review by TGWDB — APPROVED** [review]

**Pull request merged**
